File: python/infinicore/ops/matmul.py
from infinicore.lib import _infinicore
from infinicore.tensor import Tensor


def matmul(input, other, *, alpha=1.0, out=None):
    if out is None:
        return Tensor(_infinicore.matmul(input._underlying, other._underlying, alpha))

    _infinicore.matmul_(out._underlying, input._underlying, other._underlying, alpha)

    return out

File: ops/src/infinicore/ops/matmul/matmul.cc
#include "infinicore/ops/matmul.hpp"
#include "infinicore/ops/gemm.hpp"

namespace infinicore::op {

Tensor matmul(Tensor a, Tensor b, float alpha) {
    return gemm(a, b, alpha, 0.0f);
}

void matmul_(Tensor c, Tensor a, Tensor b, float alpha) {
    Gemm::execute(c, a, b, alpha, 0.0f);
}
} // namespace infinicore::op
